src/App.js

```diff
@@ -7,27 +7,44 @@ import Navigation from './components/Navigation';
 import Products from './components/Products';
 import ShoppingCart from './components/ShoppingCart';
 
+//Context APIs:
+import { ProductContext } from './contexts/ProductContext';
+import { CartContext } from './contexts/CartContext'
+
 function App() {
 	const [products] = useState(data);
 	const [cart, setCart] = useState([]);
 
 	const addItem = item => {
-		// add the given item to the cart
+		setCart([
+			...cart,
+			item
+		])
 	};
 
+	const removeItem = itemId => {
+		setCart(cart.filter(book => {
+			return book.id !== itemId
```

Comment:
With filter we'll get an array of books or items. Based on your code here are you getting an array of items or an array of booleans? It looks like it's returning the boolean value of the expression bok.id !== itemId